homeassistant/components/homematicip_cloud/__init__.py

```diff
@@ -356,6 +347,14 @@ def _get_home(hapid: str) -> Optional[AsyncHome]:
 
     if not await hap.async_setup():
         return False
+    
+    async def async_reset_hap_connection():
+        """Reset hmip hap connections."""
+        await hap.async_reset()
+        _LOGGER.debug("Shut down access point id %s", hap.config_entry.unique_id)
+
+    # Register on HA stop event to gracefully shutdown HomematicIP Cloud connection
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, async_reset_hap_connection())
```

Comment:
Store the return value of this call and unsubscribe in `async_unload_entry`.

Reply:
Is that really required for one_time_listeners? Looks like they are removed instantly.

Reply:
It's removed after the event handler is run, not before. So yes, it's still required.

Reply:
I added a new commit ("Save listener"), but i am not happy with:
It works fine if i just remove the integration, and the listener gets unregistered when the entry is unloaded.
BUT it doesn't work if i just shutdown HA. The unregister of the listener fails with a log warning:
`[homeassistant.core] Unable to remove unknown listener <function EventBus.async_listen_once.<locals>.onetime_listener at 0x1162a77a0>` because the listener is already removed before.
Any idea what to do?

Reply:
Maybe don't use `async_unload_entry` as callback? We just need to shut down the connection, right?

Reply:
That solves my problem. I had it before, but i thought async_unload_entry might be better.